### PR TITLE
Rob/Liam/Han - Added logging, fixed database up probe, so when databa…

### PR DESCRIPTION
**Pull request by hanfak**
We had a first instance of a database outage. Patterdale did not report this outage through the metrics for database_up probe. This pull request contains the fix for this.

### CODE REVIEW

File: src/main/java/io/github/tjheslin1/patterdale/database/hikari/HikariDataSourceProvider.java [hunk 1]
```diff
@@ -58,13 +56,12 @@ public static HikariDataSource retriableDataSource(PatterdaleRuntimeParameters r
...
 
-    private static HikariDataSource dataSource(PatterdaleRuntimeParameters runtimeParameters, DatabaseDefinition databaseDefinition, Passwords passwords, Logger logger)
-            throws SQLException {
+    private static HikariDataSource dataSource(PatterdaleRuntimeParameters runtimeParameters, DatabaseDefinition databaseDefinition, Passwords passwords, Logger logger) {
         try {
             String password = passwords.byDatabaseName(databaseDefinition.name).value;
             return new HikariDataSource(jdbcConfig(runtimeParameters, databaseDefinition, password));
         } catch (Exception e) {
-            logger.error("Error occurred initialising Oracle and Hikari data sources.");
+            logger.error("Error occurred initialising Oracle and Hikari data sources.", e);
```
Comment (tjheslin1) [inline src/main/java/io/github/tjheslin1/patterdale/database/hikari/HikariDataSourceProvider.java:64]: Good catch! 🤦‍♂️

**Comment by tjheslin1** [comment]
Looks good.

I will investigate further why Travis is having issue talking to oracle, for this I've spun it up locally against 2 oracle databases and it's all good.

One thing I spotted is that the new `DatabaseDownTest` fails sometimes fails if I run `./gradlew clean test` as opposed to `./gradlew test`, reporting `1.0` instead of `-1.0`. If it's not easily solvable I'll raise an issue.

**Comment by theangrydev** [comment]
The `DatabaseDownTest` is failing because of a race condition. There is a sleep where we are trying to wait for the h2 database to actually be down and for the patterdale cache to be cleared. I can bump up the sleep but not an ideal solution. I will raise an issue to see if you can think of a better approach.

**Comment by Thomas-Heslin** [comment]
Ah yes `startPatterdale` in `TestUtils` should start the app without a cache (set to `0`). I'll have a look at the issue 👍

**Pull request merged**